rescheduling-service/src/services/offerService.ts

```diff
@@ -0,0 +1,31 @@
+import { PrismaClient } from '@prisma/client';
+import fetchAvailableAppointments from './appointmentService';
+import fetchAvailablePatients from './patientService';
+
+const prisma = new PrismaClient();
+
+const createOffer = async () => {
+  try {
+    const availableAppointments = await fetchAvailableAppointments();
+    const availablePatients = await fetchAvailablePatients();
+
+    if (availableAppointments.length > 0 && availablePatients.length > 0) {
+      const selectedAppointment = availableAppointments[0];
+      // Insert much better logic to find a patient
+      const selectedPatient = availablePatients[0];
+      const offer = await prisma.offer.create({
+        data: {
+          appointment_id: selectedAppointment.id,
+          patient_id: selectedPatient.id,
+          status: 'pending',
+        },
+      });
+
+      return offer;
+    }
+  } catch (error) {
+    throw new Error('No available appointments or patients to create an offer');
+  }
+};
```

Comment:
Also, I think it's a good idea to separate this logic in the controller, so the controller will call the database to fetch the necessary data and pass this data to the createOffer service, which will handle the filter logic and give back the data, which you can then pass to another db-service call in the controller.

In that way, you will separate all the logic into components with a single responsibility.
I hope it makes sense.